src/Ed25519VerificationKey2018.js

```diff
@@ -0,0 +1,459 @@
+          asn1.fromDer(new ByteBuffer(privateKey)));
+
+        return new Ed25519VerificationKey2018({
+          publicKeyBase58: bs58.encode(publicKeyBytes),
+          // private key is the 32 byte private key + 32 byte public key
+          privateKeyBase58: bs58.encode(Buffer.concat(
+            [privateKeyBytes, publicKeyBytes])),
+          ...options
+        });
+      }
+      // create a key from the provided seed
+      const {seed} = options;
+      let seedBytes;
+      if(seed instanceof Uint8Array || Buffer.isBuffer(seed)) {
+        seedBytes = Buffer.from(seed);
+      }
+      if(!(Buffer.isBuffer(seedBytes) && seedBytes.length === 32)) {
+        throw new TypeError('`seed` must be a 32 byte Buffer or Uint8Array.');
+      }
+
+      // create a node private key
+      const privateKey = _privateKeyNode12.create({seedBytes});
+
+      // create a node public key from the private key
+      const publicKey = createPublicKey(privateKey);
+
+      // export the keys and extract key bytes from the exported DERs
+      const publicKeyBytes = publicKeyFromAsn1(
+        asn1.fromDer(new ByteBuffer(publicKey.export(publicKeyEncoding))));
+      const {privateKeyBytes} = privateKeyFromAsn1(
+        asn1.fromDer(new ByteBuffer(privateKey.export(privateKeyEncoding))));
+
+      return new Ed25519VerificationKey2018({
+        publicKeyBase58: bs58.encode(publicKeyBytes),
+        // private key is the 32 byte private key + 32 byte public key
+        privateKeyBase58: bs58.encode(Buffer.concat(
+          [privateKeyBytes, publicKeyBytes])),
+        ...options
+      });
+    }
+
+    const generateOptions = {};
+    if('seed' in options) {
+      generateOptions.seed = options.seed;
+    }
+    const {publicKey, privateKey} = ed25519.generateKeyPair(generateOptions);
+    return new Ed25519VerificationKey2018({
+      publicKeyBase58: base58.encode(publicKey),
+      privateKeyBase58: base58.encode(privateKey),
+      ...options
+    });
+  }
+  /**
+   * Creates an Ed25519 Key Pair from an existing serialized key pair.
+   * @example
+   * > const keyPair = await Ed25519VerificationKey2018.from({
+   *     controller: 'did:ex:1234',
+   *     type: 'Ed25519VerificationKey2018',
+   *     publicKeyBase58,
+   *     privateKeyBase58
+   *   });
+   *
+   * @returns {Promise<Ed25519VerificationKey2018>} An Ed25519 Key Pair.
+   */
+  static async from(options) {
+    return new Ed25519VerificationKey2018(options);
+  }
+
+  /* eslint-disable max-len */
+  /**
+   * Returns a signer object for use with
+   * [jsonld-signatures]{@link https://github.com/digitalbazaar/jsonld-signatures}.
+   * @example
+   * > const signer = keyPair.signer();
+   * > signer
+   * { sign: [AsyncFunction: sign] }
+   * > signer.sign({data});
+   *
+   * @returns {{sign: Function}} A signer for the json-ld block.
+   */
+  /* eslint-enable */
+  signer() {
+    return ed25519SignerFactory(this);
+  }
+
+  /* eslint-disable max-len */
```

Comment:
@davidlehn, @mattcollier -- what do you guys think of all these eslint exception directives?

Reply:
I removed most of those directives; no prob.